src/components/structures/TimelinePanel.js

```diff
@@ -496,10 +496,15 @@ var TimelinePanel = React.createClass({
         if (!MatrixClientPeg.get()) return;
 
         var currentReadUpToEventId = this._getCurrentReadReceipt(true);
-        var currentReadUpToEventIndex = this._indexForEventId(currentReadUpToEventId);
+        // Take the minimum of both in case they somehow get out of sync where the RM is
```

Comment:
>I also noticed that in some circumstances the RM wasnt updated because `currentReadUpToEventIndex` didnt consider the RM event ID. This was also fixed.

(From the commit message of e093702)

I'm not entirely sure how this happened but at one point I was doing some testing and noticed that if the RR was on an event which is not paginated, then it could prevent an RM being sent by returning here: https://github.com/matrix-org/matrix-react-sdk/pull/837/files/c6ccebad512f056804d71214ea003e68d91b06b0#diff-9dd3a0bf58495df3165ed62d2237e0eeL516